source/server/listener_impl.cc

```diff
@@ -26,19 +26,54 @@
 namespace Envoy {
 namespace Server {
 
-ListenSocketFactoryImplBase::ListenSocketFactoryImplBase(
-    ListenerComponentFactory& factory, Network::Address::InstanceConstSharedPtr local_address,
-    Network::Address::SocketType socket_type, const Network::Socket::OptionsSharedPtr& options,
-    bool bind_to_port, const std::string& listener_name)
-    : factory_(factory), local_address_(local_address), socket_type_(socket_type),
-      options_(options), bind_to_port_(bind_to_port), listener_name_(listener_name) {}
-
-Network::SocketSharedPtr ListenSocketFactoryImplBase::createListenSocketAndApplyOptions() {
+ListenSocketFactoryImpl::ListenSocketFactoryImpl(ListenerComponentFactory& factory,
+                                                 Network::Address::InstanceConstSharedPtr address,
+                                                 Network::Address::SocketType socket_type,
+                                                 const Network::Socket::OptionsSharedPtr& options,
+                                                 bool bind_to_port,
```

Comment:
Can you use the same creation param struct here? I might just change `duplicate_parent_socket` to `reuse_port` there for clarity? Up to you.